### PR TITLE
Add FindStormLib.cmake to handle StormLib like other dependencies.

### PR DESCRIPTION
**Pull request by mewmew**
The contents of FindStormLib.cmake is copied without modification
from the Wargus project [[1](https://github.com/Wargus/wargus/blob/master/cmake/modules/FindStormLib.cmake)].

### CODE REVIEW

**Comment by mewmew** [comment]
Output from cmake on a system with StormLib installed through the package manager. More specifically, having installed [stormlib from the AUR](https://aur.archlinux.org/packages/stormlib/) on Arch Linux.

```
-- The C compiler identification is GNU 7.1.1
-- The CXX compiler identification is GNU 7.1.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for Q_WS_X11
-- Looking for Q_WS_X11 - found
-- Looking for Q_WS_WIN
-- Looking for Q_WS_WIN - not found
-- Looking for Q_WS_QWS
-- Looking for Q_WS_QWS - not found
-- Looking for Q_WS_MAC
-- Looking for Q_WS_MAC - not found
-- Found Qt4: /usr/bin/qmake-qt4 (found suitable version "4.8.7", minimum required is "4.8") 
-- Found SDL2_mixer: /usr/lib/libSDL2_mixer.so  
-- Found Doxygen: /usr/bin/doxygen (found suitable version "1.8.13", minimum required is "1.8.0") 
-- Found BZip2: /usr/lib/libbz2.so (found version "1.0.6") 
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found StormLib: /usr/lib/libstorm.so
-- Configuring done
-- Generating done
-- Build files have been written to: /home/u/Desktop/qtd1/build
```

**Comment by mewmew** [comment]
With #2 and #3 applied I was able to successfully compile the QtD1 project. All the test cases passed and the game launches perfectly.

Thanks for sharing this project with the world! :)

Screenshoft from QtD1

**Comment by mewmew** [comment]
@aprobinson Would you mind testing this out on Debian? Install StormLib using `sudo apt-get install stormlib`, or something along those lines and try to compile with this PR.

**Comment by aprobinson** [comment]
@mewmew, @seanrobinson1114 and I are glad it's working for you! I'm looking into the sudo apt-get install command for stormlib on Debian. I'll get back to you when I've tested it. We're currently working on state machines for the actors and NPCs (check out the state_machine branch if you're interested).

File: README.md [hunk 1]
```diff
@@ -20,23 +21,21 @@ Instructions for other flavors and other OSs will be provided in the future.
 4. run `sudo apt-get install zlib1g-dev`
 5. run `sudo apt-get install libbz2-dev`
 6. run `sudo apt-get install doxygen`
+7. run `sudo apt-get install stormlib`
```
Comment (aprobinson) [inline README.md:25]: stormlib appears to be a third-party package. Please add a note along these lines: If the stormlib package cannot be found got to [this](http://xpam.pl/blog/?p=117) website and follow the instructions for deb based distros.

**Comment by aprobinson** [comment]
@mewmew, See my code comment regarding my testing of `sudo apt-get install stormlib`. Then I'll close this pull request.

**Comment by mewmew** [comment]
> @mewmew, See my code comment regarding my testing of sudo apt-get install stormlib. Then I'll close this pull request.

Done as of https://github.com/seanrobinson1114/QtD1/pull/3/commits/527496c7474c3afc79df431968f9d532e785d9d4

**Comment by mewmew** [comment]
> We're currently working on state machines for the actors and NPCs (check out the state_machine branch if you're interested).

I think using QT as the basis for GUI is a great decision, and it's exciting to see that you've started to implement the basis of game interaction.

The state_machine branch compiles well on my machine and I was able to run the game. It was possible to initiate walk actions to the player. The background music was working well (not sure if this was part of the *master* branch?)

I understand that the work in state_machine is under way :)

Quick notes. A* seems not to be implemented, so the walk actions are a bit hacky. Still, a good start. Same goes for collision detection (i.e. incorporation of the information from .SOL files). I'm sure you already have these in mind, so will be happy to try it out and provide further feedback at later stages of development.

Tomorrow, I'll head out trekking in Norway and should be back in a month or so. I'll be happy to see where you've taken it by the end of summer.

@aprobinson and @seanrobinson1114 I wish you both a lovely summer! :)

Cheers /u

**Comment by aprobinson** [comment]
Yes, everything in the state_machine branch is still a first cut as @seanrobinson1114 and I were experimenting with the state machine framework in qt to ensure that it was going to meet our needs. We're happy with what the framework provides so we're going to continue to iron out the bugs and implement collision detection throughout the summer. 

Enjoy your trekking! We're both jealous :)

Best

**Pull request merged**